lib/ohai/plugins/linux/platform.rb

```diff
@@ -28,16 +28,33 @@ def get_redhatish_version(contents)
     contents[/Rawhide/i] ? contents[/((\d+) \(Rawhide\))/i, 1].downcase : contents[/release ([\d\.]+)/, 1]
   end
 
+  # Cisco's Nexus 3/9ks are based on Wind River Linux 5. They also have a
+  # guestshell based on CentOS 7. The file /etc/os-release is read to determine
+  # the platform_family and platform, it may point to another
+  # /etc/shared/os-release file if we are within the guestshell.
   def os_release_file_is_cisco?
     return false unless File.exist?('/etc/os-release')
+    # parse /etc/os-release into a hash
     os_release_info = File.read('/etc/os-release').split.inject({}) do |map, key_value_line|
       key, _separator, value = key_value_line.partition('=')
-      map[key] = value
+      map[key] = value.gsub(/\A"|"\Z/, '') # strip leading/trailing quotes
       map
     end
-    if os_release_info['CISCO_RELEASE_INFO'] && File.exist?(os_release_info['CISCO_RELEASE_INFO'])
-      os_release_info
+    # if it has a CISCO_RELEASE_INFO value, must be Cisco
+    cisco_release_info = os_release_info['CISCO_RELEASE_INFO']
+    if cisco_release_info && File.exist?(cisco_release_info)
+      if cisco_release_info == '/etc/os-release'
+        return os_release_info
+      end
+      # read /etc/shared/os-release to override /etc/os-release info.
+      cisco_release_info = File.read(cisco_release_info).split.inject({}) do |map, key_value_line|
+        key, _separator, value = key_value_line.partition('=')
+        map[key] = value.gsub(/\A"|"\Z/, '')
+        map
+      end
+      os_release_info.merge!(cisco_release_info)
     else
+      # has /etc/os-release, but not a Cisco box
       false
     end
   end
```

Comment:
So this method is hella ugly.

First thing you probably want is a single method which reads an os-release file and renders it into a hash.

Then you probably want to write another method which does the read-os-release; find-cisco-release-info, read cisco-release-info; merge-with-os-release-info dance and then memoizes the result.  This will then be a reusable method that other stuff that inspects os-release-info can use.

Then the `os_release_file_is_cisco?` check should be vastly simplified into something like just checking `os_release_info.key?('CISCO_RELEASE_INFO')`

I'd also suggest not worrying about special casing when cisco_release_info is also '/etc/os-release' since the result of merge!'ing /etc/os-release onto /etc/os-release is fine.